projects/cmake/build.sh

```diff
@@ -153,12 +152,12 @@ fi
     fi
 
     echo "Running './regenerate.sh $(pwd)/$BUILD_DIR"
-    ./regenerate.sh $(pwd)/$BUILD_DIR
+    bash regenerate.sh $(pwd)/$BUILD_DIR
     cd ${BUILD_DIR}
     echo
     echo
     echo "Running 'cmake ../../../src $cmake_args' in $(pwd)"
-    cmake ../../../src $cmake_args
+    cmake -D Boost_NO_BOOST_CMAKE:BOOL=ON ../../../src $cmake_args
```

Comment:
what's the rationale here?  Do we want to do this for everybody?

Reply:
Not quite sure but this was necessary for me and a few other people to keep the compile working smoothless without any changes in how the compile is called. This might need a bit more discussion ...

Reply:
Yeah, more discussion might be good.  We should definitely fix the scripts so that it works smoothly for you and the few other people -- I'm worried that this would maybe break it for people that it is currently working for.

Let's figure out why the `-D Boost_NO_BOOST_CMAKE:BOOL=ON` was necessary.  It is possible that this might depend on the version of boost -- i.e. with the latest version this might not be needed.

Reply:
Yeah, I thought this was the purpose of `$cmake_args`? It allows for user-specific build arguments that may depend on their local environment (i.e. different Boost versions, install locations, CMake versions, etc).

Reply:
It's probably safe to use module mode by default though, so if we want to keep this, I'd suggest just putting `set(Boost_NO_BOOST_CMAKE ON)` into `CMakeLists.txt`

Reply:
OK, this has been reverted and pushed to a separate branch.